### DISCUSSION TITLE
Setting layer extents via API

### DISCUSSION THREAD

**Discussion by JosephTRichardson** [pelicanmapping/osgearth#2181] [Support Q&A]
Hi, I asked a question a couple weeks ago about visualizing elevation, the answer to which proved to be GDALDEM color-relief layers.

I can implement those through .earth files to good effect, but I'm now trying to do it through the osgEarth API. I can successfully create a GDALDEM layer, but when I add it to the map, the area of the elevation layer is correctly shaded -- but so is the rest of the earth, with a null value. What I get is a globe shaded in blue (0), with the area of the DTED tile shaded appropriately. How do I set the extent of the GDALDEM layer so only the area of the elevation layer is shaded?
```
std::string pathToDTEDTile = strDataDir + "/n34.dt0";
std::string pathToColorTable = strDataDir + "/colortable.txt";
osgEarth::GDALDEMLayer* gdaldem = new osgEarth::GDALDEMLayer();
gdaldem->setElevationLayer(elevationLayer);
gdaldem->setName("w086n34_relief");
gdaldem->setColorFilename(pathToColorTable.c_str());
gdaldem->setProcessing("color-relief");
gdaldem->setOpacity(0.75);
auto extents = elevationLayer->getDataExtents();
gdaldem->setDataExtents(extents);
map->addLayer(gdaldem);
```
Any idea what I'm missing?

**Comment by emminizer** [comment]
In our code we also set alpha to true:

```
  gdaldem->setAlpha(true);
```

Does that perhaps help? I don't see anything else in the initialization that looks any different from what we do, except the `setAlpha()` call. Note, this is in addition to the `setOpacity()` that you already have.
